### PR TITLE
cmake: Fix linker warning LNK4291 when linking static lib to the final binary

### PR DESCRIPTION
**Pull request by mludha**
Apparently some functions (e.g. nghttp3_balloc_clear) create xdata sections in the lib. These are used on 64bit platforms to judge which control flow transitions should be allowed. Normally they are needed only in uncommon cases like longjmp, SEH exceptions or SetThreadContext Windows API calls. These functions don't do anything to warrant it. This flag should just let the compiler do it's thing more efficiently.

### CODE REVIEW

**Comment by tatsuhiro-t** [comment]
It looks like these flags should be for x86_64 only.

**Comment by mludha** [comment]
They are necessary for x86_64 and silently ignored elsewhere. Shall I add them only for that platform?

**Comment by tatsuhiro-t** [comment]
See this error message: https://github.com/ngtcp2/nghttp3/actions/runs/9940936525/job/27555228972?pr=237

> LINK : fatal error LNK1246: '/GUARD:EHCONT' not compatible with 'x86' target machine; link without '/GUARD:EHCONT'

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,6 +103,13 @@ endif()
 set(WARNCFLAGS)
 set(WARNCXXFLAGS)
 if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+  if (CMAKE_C_COMPILER_ARCHITECTURE_ID MATCHES "x64")
```
Comment (tatsuhiro-t) [inline CMakeLists.txt:106]: Remove extra white space between "if" and "(" for consistency.

File: CMakeLists.txt [hunk 2]
```diff
@@ -103,6 +103,13 @@ endif()
...
+    set(CMAKE_C_FLAGS "/guard:ehcont ${CMAKE_C_FLAGS}")
+  endif()
+  if (CMAKE_CXX_COMPILER_ARCHITECTURE_ID MATCHES "x64")
```
Comment (tatsuhiro-t) [inline CMakeLists.txt:109]: ditto

**Review by tatsuhiro-t — CHANGES_REQUESTED** [review]

**Comment by tatsuhiro-t** [comment]
LGTM.  Thank you.  Could you squash commits into one?

**Review by tatsuhiro-t — APPROVED** [review]

**Comment by tatsuhiro-t** [comment]
Thank you for PR. Merged now.

**Pull request merged**